src/logql.grammar

```diff
@@ -76,6 +76,8 @@ LineFilter {
 Filter {
   !eql PipeExact |
   !eql PipeMatch |
+  !eql PipeMatchPattern |
+  !eql PipeNotMatchPattern |
```

Comment:
```suggestion
  !eql PipePattern |
  !eql Npa |
```

For consistency, we use the same names used by the Loki expression parser: https://github.com/grafana/loki/blob/main/pkg/logql/syntax/expr.y#L241